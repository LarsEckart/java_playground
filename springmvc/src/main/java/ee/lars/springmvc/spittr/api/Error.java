package ee.lars.springmvc.spittr.api;

public class Error {

    private final int code;
    private final String message;

    public Error(int code, String message) {
        this.code = code;
        this.message = message;
    }

    public int getCode() {
        return this.code;
    }

    public String getMessage() {
        return this.message;
    }
}