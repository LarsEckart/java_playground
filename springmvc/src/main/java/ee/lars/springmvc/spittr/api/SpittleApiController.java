package ee.lars.springmvc.spittr.api;

import ee.lars.springmvc.spittr.Spittle;
import ee.lars.springmvc.spittr.data.SpittleRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/spittles")
public class SpittleApiController {

    private static final String MAX_LONG_AS_STRING = "9223372036854775807";

    private final SpittleRepository spittleRepository;

    @Autowired
    public SpittleApiController(SpittleRepository spittleRepository) {
        this.spittleRepository = spittleRepository;
    }

    @RequestMapping(method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
    public List<Spittle> spittles(
            @RequestParam(value = "max", defaultValue = MAX_LONG_AS_STRING) long max,
            @RequestParam(value = "count", defaultValue = "20") int count)
    {
        return this.spittleRepository.findSpittles(max, count);
    }

    @RequestMapping(method = RequestMethod.POST, consumes = "application/json")
    public Spittle saveSpittle(@RequestBody Spittle spittle) {
        return spittleRepository.save(spittle);
    }

    @RequestMapping(value = "/{id}", method = RequestMethod.GET)
    public ResponseEntity<?> spittleById(@PathVariable long id) {
        Spittle spittle = this.spittleRepository.findOne(id);
        if (spittle == null) {
            Error error = new Error(4, "Spittle [" + id + "] not found");
            return new ResponseEntity<Error>(error, HttpStatus.NOT_FOUND);
        }
        return new ResponseEntity<Spittle>(spittle, HttpStatus.OK);
    }
}
