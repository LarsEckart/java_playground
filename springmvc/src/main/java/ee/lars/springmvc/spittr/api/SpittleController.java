package ee.lars.springmvc.spittr.api;

import ee.lars.springmvc.spittr.Spittle;
import ee.lars.springmvc.spittr.data.SpittleRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;

import java.util.List;

@Controller
@RequestMapping("/spittles")
public class SpittleController {

    private static final String MAX_LONG_AS_STRING = "9223372036854775807";

    private final SpittleRepository spittleRepository;

    @Autowired
    public SpittleController(SpittleRepository spittleRepository) {
        this.spittleRepository = spittleRepository;
    }

    @RequestMapping(method = RequestMethod.GET)
    public List<Spittle> spittles(
            @RequestParam(value = "max", defaultValue = MAX_LONG_AS_STRING) long max,
            @RequestParam(value = "count", defaultValue = "20") int count)
    {
        return this.spittleRepository.findSpittles(max, count);
    }
}
