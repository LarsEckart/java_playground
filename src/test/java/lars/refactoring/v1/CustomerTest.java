package lars.refactoring.v1;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class CustomerTest {

  @Test
  void statement() {
    Customer customer = new Customer("Any");

    customer.addRental(new Rental(new Movie("Harry Potter", Movie.CHILDRENS), 3));
    customer.addRental(new Rental(new Movie("Harry Potter II", Movie.CHILDRENS), 4));
    customer.addRental(new Rental(new Movie("Die Hard", Movie.REGULAR), 2));
    customer.addRental(new Rental(new Movie("Die Hard II", Movie.REGULAR), 1));
    customer.addRental(new Rental(new Movie("Die Hard III", Movie.REGULAR), 3));
    customer.addRental(
        new Rental(new Movie("Fantastic BBeasts And Where To Find Them", Movie.NEW_RELEASE), 4));

    assertThat(customer.statement())
        .isEqualTo(
            "Rental Record for Any\n"
                + "\tHarry Potter\t1.5\n"
                + "\tHarry Potter II\t3.0\n"
                + "\tDie Hard\t2.0\n"
                + "\tDie Hard II\t2.0\n"
                + "\tDie Hard III\t3.5\n"
                + "\tFantastic BBeasts And Where To Find Them\t12.0\n"
                + "Amount owed is 24.0\n"
                + "Your earned 7 frequent renter points");
  }
}
