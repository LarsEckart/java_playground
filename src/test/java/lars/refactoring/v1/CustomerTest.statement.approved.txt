Rental Record for Any
	Harry Potter	1.5
	Harry Potter II	3.0
	Die Hard	2.0
	Die Hard II	2.0
	Die Hard III	3.5
	Fantastic Beasts And Where To Find Them	12.0
Amount owed is 24.0
Your earned 7 frequent renter points
